packages/replay-next/src/hooks/usePositionedTooltip.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Dispatch,
+  MouseEvent,
+  ReactNode,
+  RefObject,
+  SetStateAction,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+
+import Tooltip from "replay-next/components/Tooltip";
+
+import { MARGIN_LARGE, MARGIN_SMALL, TooltipPosition } from "./useTooltip";
+
+// A copy-pasted version of the other `useTooltip` hook,
+// but exposing `setShowTooltip` for control from the parent component
+// instead of controlling display itself on hover.
+// TODO Find a better way to consolidate these or extract common logic
+export function usePositionedTooltip({
+  className,
+  containerRef,
+  position,
+  tooltip,
+  targetRef,
+}: {
+  className?: string;
+  containerRef?: RefObject<HTMLElement>;
+  position: TooltipPosition;
+  targetRef: RefObject<HTMLElement>;
+  tooltip: ReactNode;
+}): {
+  setShowTooltip: Dispatch<SetStateAction<boolean>>;
+  tooltip: ReactNode;
+} {
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  const [showTooltip, setShowTooltip] = useState(false);
```

Comment:
I think sharing state inside and outside of the hook this way is still a bit odd, but I can see why you do it with the hide-on-scroll behavior. 🤷🏼